src/util/std_expr.h

```diff
@@ -864,32 +815,29 @@ class multi_ary_exprt:public exprt
   }
 };
 
-/*! \brief Cast a generic exprt to a \ref multi_ary_exprt
- *
- * This is an unchecked conversion. \a expr must be known to be \ref
- * multi_ary_exprt.
- *
- * \param expr Source expression
- * \return Object of type \ref multi_ary_exprt
- *
- * \ingroup gr_std_expr
-*/
+/// \brief Cast a generic exprt to a \ref multi_ary_exprt
+///
+/// This is an unchecked conversion. \a expr must be known to be \ref
+/// multi_ary_exprt.
+///
+/// \param expr Source expression
+/// \return Object of type \ref multi_ary_exprt
+///
+/// \ingroup gr_std_expr
 inline const multi_ary_exprt &to_multi_ary_expr(const exprt &expr)
 {
   return static_cast<const multi_ary_exprt &>(expr);
 }
 
-/*! \copydoc to_multi_ary_expr(const exprt &)
- * \ingroup gr_std_expr
-*/
+/// \copydoc to_multi_ary_expr(const exprt &)
+/// \ingroup gr_std_expr
 inline multi_ary_exprt &to_multi_ary_expr(exprt &expr)
 {
   return static_cast<multi_ary_exprt &>(expr);
 }
 
 
-/*! \brief The plus expression
-*/
+/// \brief The plus expression
```

Comment:
What is the semantics of a multi-ary plus? Is it right or left associative?

Reply:
Uh, looked like a binary plus to me (as we can only construct it using two operands). But it's true we can obtain multi-ary plus by cast and the number of arguments is then not restricted.
Judging by the implementation of the SMT2, it seems left-associative, (https://github.com/diffblue/cbmc/blob/develop/src/solvers/smt2/smt2_conv.cpp#L2946), but I don't know where we do the flattening of `plus_exprt` for the SAT solver.

Reply:
I believe we do not place any constraints here, and indeed assume that `a + b + c == (a + b) + c == a + (b + c)`. To that effect, different implementations of evaluation (e.g., the simplifier vs. solver back-ends) may differ in their evaluation order and thus in the associativity they pick.

Reply:
Added description mentioning that no associativity is specified on the operator.